src/transformers/modeling_tf_utils.py

```diff
@@ -1138,6 +1195,8 @@ def _generate_beam_search(
             assert (len(hypo) == max_length for hypo in best)
             decoded = tf.stack(best)
 
+        if self.config.is_encoder_decoder and decoder_start_token_id != bos_token_id:
```

Comment:
@thomwolf @sshleifer This if statement might be debatable:
- In my opinion it is a special behavior (of Bart) to cut the first decoded output id and it should only happen if a special `decoder_start_token_id` is set (which happens for Bart).

Reply:
I'm definitely not a fan of having model-specific behavior in the base class.

Shall we have a `filter_generation()` method that would be overridden by Bart only for now?

Reply:
Actually, removed this if statement completely. 

Think this is the best solution, because:

When decoding bart output with `remove_special_characters=True` as is done in all 
tests and examples, the final decoded solution is the same as before.